vtm/src/org/oscim/layers/tile/buildings/BuildingLayer.java

```diff
@@ -73,18 +75,34 @@ public BuildingLayer(Map map, VectorTileLayer tileLayer, boolean mesh) {
     }
 
     public BuildingLayer(Map map, VectorTileLayer tileLayer, int zoomMin, int zoomMax, boolean mesh) {
-
         super(map);
 
         tileLayer.addHook(this);
 
-        mRenderer = new BuildingRenderer(tileLayer.tileRenderer(),
-                zoomMin, zoomMax,
+        // Use zoomMin as zoomLimit to render buildings only once
+        this.setZoomLimiter(new ZoomLimiter(tileLayer.getManager(), zoomMin, zoomMax, zoomMin));
+
+        mRenderer = new BuildingRenderer(tileLayer.tileRenderer(), mZoomLimiter,
                 mesh, !mesh && TRANSLUCENT); // alpha must be disabled for mesh renderer
         if (POST_AA)
             mRenderer = new OffscreenRenderer(Mode.SSAO_FXAA, mRenderer);
     }
 
+    @Override
+    public void setZoomLimiter(ZoomLimiter zoomLimiter) {
+        mZoomLimiter = zoomLimiter;
+    }
```

Comment:
`mZoomLimiter` is already used in other classes, so cannot(?) change it later via a setter.

Reply:
Yes, seems needless as `ZoomLimiter` never would be set outside class which implements `IZoomLimiter`.